File: pathsum3.js
const dfs = (node, t, sum, count, trail) => {
  if(!node) return;
  let tr = [...trail, node.val], i = tr.length, ;
  while(i>=0)
  if(sum + node.val === t) count++;
  if(node.left){
      count = dfs(node.left, t, sum+node.val, count)
  }
  if(node.right){
      count = dfs(node.right, t, sum+node.val, count)
  }
  return count;
}
var pathSum = function(root, sum) {
// if(!root) return 0;
  return dfs(root, sum, 0, 0)
};

File: pathsum3.py
class Solution:
    def pathSum(self, root, k):
        
        def traverse(node, table, running):
            if not node:
                return 0
            running += node.val
            path = table[running - k] if running-k in table else 0
            table[running] = table[running]+1 if running in table else 1
            path += traverse(node.left, table, running) + traverse(node.right, table, running)
            table[running] -= 1
            
            return path
        
        return traverse(root, {0: 1}, 0)